tool-plugins/intellij/src/main/java/io/ballerina/plugins/idea/debugger/BallerinaDAPClientConnector.java

```diff
@@ -0,0 +1,224 @@
+
+    private BallerinaDebugProcess context;
+    private Project project;
+    private String host;
+    private int port;
+    private DAPClient debugClient;
+    private IDebugProtocolServer debugServer;
+    private DAPRequestManager requestManager;
+    private BallerinaStreamConnectionProvider streamConnectionProvider;
+    private Future<Void> launcherFuture;
+    private CompletableFuture<Capabilities> initializeFuture;
+    private Capabilities initializeResult;
+    private ConnectionState myConnectionState;
+
+    private static final int DEBUG_ADAPTOR_PORT = 4711;
+    private static final String CONFIG_SOURCEROOT = "sourceRoot";
+    private static final String CONFIG_DEBUGEE_PORT = "debuggeePort";
+
+    public BallerinaDAPClientConnector(@NotNull Project project, @NotNull String host, int port) {
+        this.project = project;
+        this.host = host;
+        this.port = port;
+        myConnectionState = ConnectionState.NOT_CONNECTED;
+    }
+
+    public DAPRequestManager getRequestManager() {
+        return requestManager;
+    }
+
+    public Project getProject() {
+        return project;
+    }
+
+    public int getPort() {
+        return port;
+    }
+
+    public BallerinaDebugProcess getContext() {
+        return context;
+    }
+
+    public void setContext(BallerinaDebugProcess context) {
+        this.context = context;
+    }
+
+    void createConnection() {
+        try {
+            myConnectionState = ConnectionState.CONNECTING;
+            streamConnectionProvider = createConnectionProvider(project);
+            if (streamConnectionProvider == null) {
+                LOG.warn("Unable to establish the socket connection provider.");
+                return;
+            }
+            streamConnectionProvider.start();
+            Pair<InputStream, OutputStream> streams = new ImmutablePair<>(streamConnectionProvider.getInputStream(),
+                    streamConnectionProvider.getOutputStream());
+            InputStream inputStream = streams.getKey();
+            OutputStream outputStream = streams.getValue();
+
+            if (inputStream == null || outputStream == null) {
+                LOG.warn("Unable to establish connection with the debug server.");
+            }
+
+            debugClient = new DAPClient();
+            Launcher<IDebugProtocolServer> clientLauncher = DSPLauncher.createClientLauncher(debugClient,
+                    inputStream, outputStream);
+            debugServer = clientLauncher.getRemoteProxy();
+            launcherFuture = clientLauncher.startListening();
+
+            InitializeRequestArguments initParams = new InitializeRequestArguments();
+            initParams.setAdapterID("BallerinaDebugClient");
+
+            initializeFuture = debugServer.initialize(initParams).thenApply(res -> {
+                initializeResult = res;
+                LOG.info("Debug server initialize result received.");
```

Comment:
Should this be a `info` log? and consider re-phrasing the sentence.

Reply:
If we use `debug` level, users will have to reproduce an issue instead of just sending the idea log files. Re-phrased the sentence.